src/app/parcels/ParcelsPage.tsx

```diff
@@ -434,31 +434,40 @@ const ParcelsPage: React.FC<{}> = () => {
 
     const parcelsTableFetchAbortController = useRef<AbortController | null>(null);
 
-    useEffect(() => {
-        if (parcelId) {
-            setSelectedParcelId(parcelId);
-            setModalIsOpen(true);
-        }
-    }, [parcelId]);
-
-    useEffect(() => {
-        const fetchAndSetClientIdForSelectedParcel = async (): Promise<void> => {
+    const fetchAndSetClientIdForSelectedParcel = async (): Promise<void> => {
+        if (!parcelId) {
             const { data, error } = await supabase
                 .from("parcels")
                 .select("client_id")
                 .eq("primary_key", parcelId)
                 .single();
 
             if (error) {
-                void logErrorReturnLogId("Error fetching clientId from database", { error });
-                return;
+                const logId = await logErrorReturnLogId(
+                    "Error fetching clientId from database",
+                    error
+                );
+                throw new DatabaseError("fetch", "client id", logId);
             }
 
             const fetchedClientId = data.client_id;
             setClientIdForSelectedParcel(fetchedClientId);
-        };
+        }
+        return;
+    };
+
+    useEffect(() => {
+        if (parcelId) {
+            setSelectedParcelId(parcelId);
+            setModalIsOpen(true);
+        }
+    }, [parcelId]);
+
+    useEffect(() => {
+        if (parcelId) {
+            void fetchAndSetClientIdForSelectedParcel();
+        }
         setClientIdForSelectedParcel(null);
```

Comment:
We should probably set client id to null first right?
```suggestion
        setClientIdForSelectedParcel(null);
        if (parcelId) {
            void fetchAndSetClientIdForSelectedParcel();
        }
```